kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageWriteApiBase.java

```diff
@@ -124,16 +134,39 @@ protected Map<Integer, String> getRowErrorMapping(Exception exception) {
         }
     }
 
+    protected boolean getAutoCreateTables() {
+        return this.autoCreateTables;
+    }
+
     /**
-     * Wait at least {@link #retryWait}, with up to an additional 1 second of random jitter.
-     * @throws InterruptedException if interrupted.
+     * Attempts to create table
+     * @param retryHandler Plain java object which handles retries on table operation
+     * @param tableOperation lambda of the table operation to perform
      */
-    protected void waitRandomTime() throws InterruptedException {
-        Thread.sleep(retryWait + random.nextInt(1000));
+    protected void attemptTableOperation(StorageWriteApiRetryHandler retryHandler, TableOperation tableOperation) {
```

Comment:
minor comment - maybe this responsibility resides in RetryHandler where we can just pass the operation to retry.

Reply:
Thanks for the input, moved it there.